### PR TITLE
Report Content

### LINKED ISSUE #70: [Moderation] - Report Button
As a User
I want to see a button on every user content
in order to report the specific content

### PR DESCRIPTION
**Pull request by appinteractive**
## Implement basic content reporting

As a User
I would like to click on the button 
to report inappropriate content

## Todos
- [x] Write Integration Tests 

## UI

bildschirmfoto 2019-01-11 um 17 38 31
bildschirmfoto 2019-01-11 um 17 38 37
bildschirmfoto 2019-01-11 um 19 19 47

## Dependencies

Relates to https://github.com/Human-Connection/Nitro-Backend/pull/99
Close #70

### CODE REVIEW

File: components/PostCard.vue [hunk 1]
```diff
@@ -82,6 +123,23 @@ export default {
...
+    },
+    routes() {
+      let routes = [
+        {
+          name: this.$t('common.reportContent'),
+          path: `/profile/`,
+          icon: 'flag'
+        }
+      ]
+      // if (this.isAdmin) {
```
Comment (roschaefer) [inline components/PostCard.vue]: No disabled code please

**Comment by appinteractive** [comment]
@roschaefer could you already take a look at the cucumber tests?

File: cypress/integration/05.ReportContent.feature [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  I would like to see all reported content
+  So I can look into it and decide what to do
+
+  Scenario: Report a Post from landingpage
+    Given I am logged in as "user"
+    And I am on the "landing" page
+
+    When I click on a Post menu and select the report option
+    And I click on send in the confirmation dialog
+    Then I get a success message
```
[the remark this answers is not in the document]
Comment (roschaefer) [reply cypress/integration/05.ReportContent.feature]: alternatively: `And an abuse report was created in the database`

File: cypress/integration/05.ReportContent.feature [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  Scenario: See reported content
+    Given I am logged in as "moderator"
+    And I previously reported a post
+
+    When I am on the "moderation" page
+    Then I see my reported post
+
+  Scenario: Report while reading Post
+    Given I am logged in as "admin"
```
Comment (roschaefer) [inline cypress/integration/05.ReportContent.feature]:
```suggestion
    Given I am logged in as an ordinary user without privileges
```
I would take the user with the lowest authorization.

File: cypress/integration/05.ReportContent.feature [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    And I am viewing a post
+
+    When I report the current post
+    And I visit the "moderation" page
+    Then I see my reported post
```
Comment (roschaefer) [inline cypress/integration/05.ReportContent.feature]:
```suggestion
    Then all the reported content is there
```
This communicates that all reported content is in one central location

File: cypress/integration/05.ReportContent.feature [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    When I click on the author
+    And I report the author
+    And I visit the "moderation" page
+    Then I see my reported user
+
+  Scenario: Normal user can't see the moderation page
+    Given I am logged in as "user"
+
+    When I can click on my profile picture in the top right corner
+    Then I can't see the moderation menu item
```
Comment (roschaefer) [inline cypress/integration/05.ReportContent.feature]: The absence of a UI is really hard to test. I would suggest to remove that scenario. Only test what is there, not what is not there.

From the implementation it's hard to check for non-existence, e.g.checking a blank page for text not being there is always `true`.

**Pull request merged**